Add Rancher upgrade note and MCM related changes for v1.2 (#405)
harvester/docs: docs/rancher/rancher-integration.md

Users can import and manage multiple Harvester clusters using the Rancher [Virtualization Management](virtualization-management.md) feature and leverage the Rancher's [authentication](https://ranchermanager.docs.rancher.com/v2.7/pages-for-subheaders/authentication-config) feature and [RBAC control](https://ranchermanager.docs.rancher.com/v2.7/pages-for-subheaders/manage-role-based-access-control-rbac) for [multi-tenancy](virtualization-management.md#multi-tenancy) support.

For a comprehensive overview of the support matrix, please refer to the [Harvester & Rancher Support Matrix](https://www.suse.com/suse-harvester/support-matrix/all-supported-versions/).

For the network requirements, please refer to the doc [here](install/requirements.md#network-requirements).

<div class="text-center">
<iframe width="950" height="475" src="https://www.youtube.com/embed/fyxDm3HVwWI" title="YouTube video player" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</div>

![virtualization-management](/img/v1.2/rancher/virtualization-management.png)

## Deploying Rancher server

> To use Rancher with Harvester, please install the Rancher in a separated server. If you want to try out the integration features, you can create a VM in Harvester and install Rancher server by following the [Helm CLI quick start](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/helm-cli).

Reviewer: ```suggestion
To use Rancher with Harvester, please install Rancher on a separate server. If you want to try out the integration features, you can create a VM in Harvester and install the Rancher server by following the [Helm CLI quick start](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/helm-cli).
```

- [AWS Marketplace](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/aws-marketplace) (uses Amazon EKS)
- [Azure](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/azure) (uses Terraform)
- [DigitalOcean](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/digitalocean) (uses Terraform)
- [GCP](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/gcp) (uses Terraform)
- [Hetzner Cloud](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/hetzner-cloud) (uses Terraform)
- [Vagrant](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/vagrant)
- [Equinix Metal](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/equinix-metal)
- [Outscale](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/outscale-qs) (uses Terraform)

> If you prefer, the following guide will take you through the same process in individual steps. Use this if you want to run Rancher in a different provider, on prem, or if you would just like to see how easy it is.

Reviewer 2: ```suggestion
If you prefer, the following guide will take you through the same process in individual steps. Use this if you want to run Rancher in a different provider, on prem, or if you want to see how easy it is.
```

- [Manual Install](https://ranchermanager.docs.rancher.com/v2.7/getting-started/quick-start-guides/deploy-rancher-manager/helm-cli)

:::caution

> **Do not install Rancher with Docker in production**. Otherwise, your environment may be damaged and your cluster may not be recovered. Installing Rancher in Docker should only be used for quick evaluation and testing purposes.

Reviewer 2: ```suggestion
**Do not install Rancher with Docker in production**. Otherwise, your environment may be damaged, and your cluster may not be abled to be recovered. Installing Rancher in Docker should only be used for quick evaluation and testing purposes.
```

:::

## Virtualization management

With Rancher's virtualization management feature, you can import and manage your Harvester cluster. By clicking one of the imported clusters, you can easily access and manage a range of Harvester cluster resources, including Hosts, VMs, images, volumes, and more. 
Additionally, the virtualization management feature leverages Rancher's existing capabilities, such as authentication with various auth providers and multi-tenancy support.

For in-depth insights, please refer to the [virtualization management](./virtualization-management.md) page.

![import-cluster](/img/v1.2/rancher/import-harvester-cluster.png)

## Creating Kubernetes clusters using the Harvester node driver

> You can have Rancher launch a Kubernetes cluster using the [Harvester node driver](./node/node-driver.md). When Rancher deploys Kubernetes onto these nodes, you can choose between Rancher Kubernetes Engine (RKE) or RKE2 distributions.

Reviewer: ```suggestion
You launch a Kubernetes cluster from Rancher using the [Harvester node driver](./node/node-driver.md). When Rancher deploys Kubernetes onto these nodes, you can choose between Rancher Kubernetes Engine (RKE) or RKE2 distributions.
```

One benefit of installing Kubernetes on node pools hosted by the node driver is that if a node loses connectivity with the cluster, Rancher can automatically create another node to join the cluster to ensure that the count of the node pool is as expected.

> Starting from Rancher version `v2.6.1`, the Harvester node driver is included by default. Users can refer to the [node-driver](./node/node-driver.md) page for more details.

Reviewer: ```suggestion
Starting from Rancher version `v2.6.1`, the Harvester node driver is included by default. You can refer to the [node-driver](./node/node-driver.md) page for more details.
```
